pkg/server/test/list_users.go

```diff
@@ -0,0 +1,199 @@
+			},
+			inputConfigMaxResults: 0, // infinite
+			expectedResults:       100 * 50,
+		},
+		`all_results_with_conditions`: {
+			inputModel: `model
+							schema 1.1
+						type user
+						type folder
+							relations
+								define viewer: [user]
+						type document
+							relations
+								define viewer: [user with condTrue]
+								define parent: [folder]
+								define can_view_conditional: viewer or viewer from parent
+
+						condition condTrue(param: bool) {
+							param == true
+						}`,
+			tupleGenerator: func() []*openfgav1.TupleKey {
+				var tuples []*openfgav1.TupleKey
+				for i := 0; i < 100; i++ {
+					for j := 0; j < 50; j++ {
+						// one document accessible by many users
+						user := fmt.Sprintf("user:%s", ulid.Make().String())
+						tuples = append(tuples, tuple.NewTupleKeyWithCondition("document:1", "viewer", user, "condTrue", nil))
+					}
+				}
+				return tuples
+			},
+			inputRequest: &openfgav1.ListUsersRequest{
+				Object:      &openfgav1.Object{Type: "document", Id: "1"},
+				Relation:    "can_view_conditional",
+				UserFilters: []*openfgav1.UserTypeFilter{{Type: "user"}},
+				Context:     testutils.MustNewStruct(b, map[string]interface{}{"param": true}),
+			},
+			inputConfigMaxResults: 0, // infinite
+			expectedResults:       100 * 50,
+		},
+		`exclusion_without_conditions`: {
+			inputModel: `model
+							schema 1.1
+						type user
+						type document
+							relations
+								define blocked: [user]
+								define public_but_not: [user:*] but not blocked`,
+			tupleGenerator: func() []*openfgav1.TupleKey {
+				return []*openfgav1.TupleKey{
+					tuple.NewTupleKey("document:1", "public_but_not", "user:*"),
+				}
+			},
+			inputRequest: &openfgav1.ListUsersRequest{
+				Object:      &openfgav1.Object{Type: "document", Id: "1"},
+				Relation:    "public_but_not",
+				UserFilters: []*openfgav1.UserTypeFilter{{Type: "user"}},
+			},
+			inputConfigMaxResults: 0, // infinite
+			expectedResults:       1,
+		},
+	}
+
+	for name, bm := range benchmarkScenarios {
+		b.Run(name, func(b *testing.B) {
+			ctx := context.Background()
+			storeID := ulid.Make().String()
+
+			// arrange: write model
+			model := testutils.MustTransformDSLToProtoWithID(bm.inputModel)
+			typeSystem, err := typesystem.NewAndValidate(context.Background(), model)
+			require.NoError(b, err)
+			err = ds.WriteAuthorizationModel(context.Background(), storeID, model)
+			require.NoError(b, err)
+
+			// arrange: write tuples
+			tuples := bm.tupleGenerator()
+			for i := 0; i < len(tuples); {
+				var tuplesToWrite []*openfgav1.TupleKey
+				for j := 0; j < ds.MaxTuplesPerWrite(); j++ {
+					if i == len(tuples) {
+						break
+					}
+					tuplesToWrite = append(tuplesToWrite, tuples[i])
+					i++
+				}
+				err = ds.Write(context.Background(), storeID, nil, tuplesToWrite)
+				require.NoError(b, err)
+			}
+
+			b.ResetTimer()
+
+			for i := 0; i < b.N; i++ {
+				bm.inputRequest.StoreId = storeID
```

Comment:
Any reason why we're including the setup code in the benchmark run?

Shouldn't we setup the `NewListUsersQuery` externally and then just benhcmark the `ListUsers` call directly?

Reply:
💀 🤦‍♀️ no reason.. moved it.